src/compat/browser_detection.ts

```diff
@@ -30,12 +30,20 @@ const isIEOrEdge : boolean = isNode ?
   /(Trident|Edge)\//.test(navigator.userAgent);
 
 const isFirefox : boolean = !isNode &&
-                            navigator.userAgent.toLowerCase().indexOf("firefox") !== -1;
+                            typeof (window as any).InstallTrigger !== "undefined";
```

Comment:
Isn't that new version riskier if `InstallTrigger` is not defined on some future versions of firefox?

MDN seems to be telling that this is a legacy API.

As the functions declared here are only to fix specific browser bugs (and not for feature detection), I prefer the "user-agent" version here. It's true that the user-agent became a mess and that we're also not protected from false positives/negatives in the future there, but it still looks less risky short term to me (and we would be using the user-agent for what it was made for).

https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent actually supports the use of the user-agent for fixing unworkable browser-specific bugs (albeit precizing that it is for "browsers that do not automatically update").

An other left problem with the user-agent technique would be that we will be vulnerable to user-agent spoofing. But considering the few bugs this constants actually fixes (2 bugs linked to native TextTracks handling and a rare bug linked to the player freezing until a seek is done), I guess we could let them for those cheeky technical-aware users.